iina/Extensions.swift

```diff
@@ -163,6 +163,16 @@ extension Array {
   }
 }
 
+extension Dictionary {
+  mutating func safeAppend<T: Equatable>(_ value: T, forKey key: Key) where Value == Array<T> {
+    if self[key] == nil {
+      self[key] = Array<T>()
```

Comment:
I'm not sure whether it would work. `var v = self[key]` will copy `self[key]` to a new array.

```swift
var dict: [String: [String]] = ["foo": ["a"]]

if var v = dict["foo"] {
  v.append("b")
  print(v.count)
  print(dict["foo"]!.count)
}

print(dict)

// 2
// 1
// ["foo": ["a"]]
```